### ISSUE TITLE
"Search ..." commands open an empty tab

### ISSUE THREAD

**Issue by Valentino1970** [Robbendebiene/Gesturefy#621]
Hi! Using a gesture to search selected text in a new tab with option highlight the new tab open an empty tab. Same problem in the last few Gesturefy versions.

- Operating system: Windows 8.1
- Firefox version: 94.0.1
- Gesturefy version: 3.2.3

**Comment by Robbendebiene** [comment]
Hi, it's working fine for me. Since when does this problem occur to you?

Can you please try to reassign the command to your gesture, save your gesture and see if it works then?

**Comment by Valentino1970** [comment]
Hi and thank you for your answer.

It started some days ago (maybe after one automatic update?). I tried everything (downgrade, inspect console, reassigning and so on) with no success.
If I reassign and save, the search run opening in a new tab and taking focus, BUT only the first search, than next ones one anf focus new blank tab.
Hopying this can help...

**Comment by Robbendebiene** [comment]
Ohh, so it's opening a completely empty tab without showing an empty search engine. I've experienced the problem too about 3 days ago on my working machine, but it only happened on one particular site and after a browser restart I wasn't able to reproduce it again. I've also checked the console and it was empty. Since downgrading Gesturefy doesn't fix the problem I assume it's some sort of bug in Firefox 94 which was released recently.

> If I reassign and save, the search runs opening in a new tab and taking focus, BUT only for the first search, than next ones only open and focus new blank tabs.

This however makes me think it's a bug in Gesturefy.

Since I'm not able to reproduce it myself I rely on further investigation by you. Maybe you can detect some sort of pattern.
Just to make sure please check the console again the following way:


1. Enter `about:devtools-toolbox?type=extension&id=%7B506e023c-7f2b-40a3-8066-bc5deb40aebe%7D` in your address bar
2. Switch to the "Console" tab in the top bar
3. Clear the console via the bin icon (because it also shows errors unrelated to Gesturefy)
4. Reproduce the error
3. Head back to the console and copy and paste all outputs into the issue

**Comment by Valentino1970** [comment]
I made it, only messages are:
16:41:22.433 Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
    sendOptions moz-extension://2e569851-022b-4929-97e5-cfa322a210cd/background.js:228
background.js:228:18
16:41:24.538 Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
    sendOptions moz-extension://2e569851-022b-4929-97e5-cfa322a210cd/background.js:228
background.js:228:18

**Comment by Robbendebiene** [comment]
Looks like this is an error from a different extension since Gesturefy's `background.js` only consists of 211 lines of code.

**Comment by Valentino1970** [comment]
Yes, I tried in the past days to unable all the extensions that could be involved, but with no success. Besides, the fact that the first time the gesture works made me think it was a gesturefy extension bug, but I'll try to unable selectively by groups all the extensions to try to solve. I'll update you if I have news, thank you

**Comment by Robbendebiene** [comment]
In theory another extension shouldn't be able to interfere with Gesturefy at least in this particular scenario. So it's probably not caused by another extension. Currently I'm able to reproduce the error on my working machine again but it only happens sometimes or on some websites. I really cannot make sense of it right now.

**Comment by Valentino1970** [comment]
Yes, it is the same for me, that is the reason I wrote to you ;)
Trying to experiment with extensions, but with crazy results: sometimes the problem seems solved (gesturefy works as espected), but if I try to reproduce the procedure to solve the problem, the problem came back...I go on to try...

**Comment by Robbendebiene** [comment]
Just to be sure, you are using the "Search Text Selection In New Tab" right? Can you confirm that it does work if you disable tab focus/highlight?

> maybe something esoteric about firefox?

Yea.. if you can confirm the above then this must be the case.

**Comment by Valentino1970** [comment]
Finally I solved! I made some esoteric research on my pc and the culprit is FISSION (site isolation) in firefox. The problem is present (with crazy results) when FISSION is enabled. I think firefox team has to work a little more on it.
I hope this will be useful to you ;)
Thank you for your help and your great extension

**Comment by Robbendebiene** [comment]
That's it, great catch! :) They just started rolling this out so it makes sense it doesn't happen on every machine. I'll open a ticket on bugzilla.

**Comment by Valentino1970** [comment]
;)

**Comment by Robbendebiene** [comment]
Just for reference: https://bugzilla.mozilla.org/show_bug.cgi?id=1741694

**Comment by Valentino1970** [comment]
Thank you

**Comment by Bahoslav** [comment]
I'm having same problem.

**Comment by Robbendebiene** [comment]
For now this can only be "solved" by going to `about:config` and setting `fission.autostart`  to `false`.

**Comment by bubblefoil** [comment]
Wow, it started happening again. So I checked the `fission.autostart` flag and apparently a recent update re-enabled it!
It's sad there's little progress regarding the bug in Firefox and they are also breaking the workaround.

**Comment by byzod** [comment]
Lucky enough to capture a log for this bug (when try to search some text with search popup menu)

```
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
2 [iframe-popup-command-view.mjs:113](moz-extension://e9a0bda0-215b-4f76-8cfa-2da37bcb8b79/core/views/popup-command-view/iframe-popup-command-view.mjs)
    terminate moz-extension://e9a0bda0-215b-4f76-8cfa-2da37bcb8b79/core/views/popup-command-view/iframe-popup-command-view.mjs:113
```

**Comment by Robbendebiene** [comment]
@byzod Thanks for your investigation. However this is not related to the problem. You can also reproduce the problem with the simple "Search selected text" command.
The error message existed before and shouldn't affect the extension in any way. I never found a way to properly suppress the error, but I remember the devs on mozilla were talking about removing it all together, because it has now use.

**Comment by ScienceDiscoverer** [comment]
It's still happening. I only learned about this extension today, I'm total noob, and got startled by this bug. I guess I'm a bug-magnet =\

I can add few interesting findings. The most interesting one: when I open console tab to try and catch some error messages, the bug disappears completely and cannot be reproduced! I'm using "Search highlighted text in new tab" with focus switch to created tab. I can search text on any site any number of times with no issue. Also nothing is written to console. However, when I close console tab and try to search text again, bug returns!

One more observation - this bug only happens if I search selected text AND have "Switch to search tab" option enabled. If searching without focus shift, there is no bug. Also with custom URL in search field, but seems to not happen too... But "empty search disable" rule does not work in this case. Yea, its a mess!

Though, I got lucky I found this issue, at least! I hope that switching off  `fission.autostart` will solve this, for now...

**Comment by OLOPOLON** [comment]
I've recently uptaded Firefox from 89.01 to 103 and later 104, and both newest versions cause that empty tab error. Sometimes it works, sometimes not. It was better on 103, cause there was some regularity in the occurrence of this problem. Every first using of pop-up particular  search engine was opened empty tab, but if you used the same search engine again, you saw results of your searching in new tab. In 104 version problem became more often and unpredictable, so I was forced back to old 89.01 version :(. BTW, is it possible to add option to select individual search engines which show in pop-up window?

**Comment by mherrmann3** [comment]
I didn't notice differences among the >v100 versions; but occasionally, empty tabs appear, yes.

@OLOPOLON regarding your "BTW" (select individual search engines): have a look here: https://github.com/Robbendebiene/Gesturefy/issues/584. Currently, this is not possible, but you may use my suggested workaround by "outsourcing" search engines to bookmarks (see last 4 comments).

**Comment by byzod** [comment]
Good news is that this bug is stable now, I ALWAYS get a blank page, the search popup is unusable completely

**Comment by Robbendebiene** [comment]
Indeed, I can confirm this. But since I'm unable to fix the bug please post any valuable information here: https://bugzilla.mozilla.org/show_bug.cgi?id=1741694

**Comment by byzod** [comment]
> Indeed, I can confirm this. But since I'm unable to fix the bug please post any valuable information here: https://bugzilla.mozilla.org/show_bug.cgi?id=1741694

[This addon](https://github.com/harytfw/GlitterDrag) has a working feature that you can select words and drag to search it with current search engine, looks similar. Hope it helps

**Comment by byzod** [comment]
Yes I can confirm it was fixed in 119b4 :D

**Comment by byzod** [comment]
Can I have a timeout hack version of gesturefy as you mentioned please?  @Robbendebiene

> Thanks for the Info, looks like the developer [added a timeout](https://github.com/harytfw/GlitterDrag/blob/main/src/background/executor.ts#L105) there as a workaround.

Since glitterdrag's drag search is 100% reliable for me while waiting Mozilla to fix this bug might take another 10 years... ;_;

**Comment by Robbendebiene** [comment]
> Can I have a timeout hack version of gesturefy as you mentioned please? @Robbendebiene

Try https://github.com/Robbendebiene/Gesturefy/archive/refs/heads/empty-tab-on-search-mitigation.zip

Extract the zip and then re-zip everything inside the `src` directory. Then you should be able to install it clicking the "gear" icon under `about:addons`

**Comment by byzod** [comment]
> > Can I have a timeout hack version of gesturefy as you mentioned please? @Robbendebiene
> 
> Try https://github.com/Robbendebiene/Gesturefy/archive/refs/heads/empty-tab-on-search-mitigation.zip
> 
> Extract the zip and then re-zip everything inside the `src` directory. Then you should be able to install it clicking the "gear" icon under `about:addons`

Thank you so much, it works like a charm ❤️ 

Never expect 50 milliseconds  would make such big difference, I love hacking

And fuck Mozilla

**Comment by happysurf** [comment]
> > Can I have a timeout hack version of gesturefy as you mentioned please? @Robbendebiene
> 
> Try https://github.com/Robbendebiene/Gesturefy/archive/refs/heads/empty-tab-on-search-mitigation.zip
> 
> Extract the zip and then re-zip everything inside the `src` directory. Then you should be able to install it clicking the "gear" icon under `about:addons`

For me is impossible install, I get the error damaged file.

**Comment by b2wood** [comment]
All preset gestures are deleted.  Rocker gesture will open a new blank tab search and sometimes it willl open a new tab with selected text. 
Turned off rocker gesture and turned on wheel gesture with wheel up. Same results but will open multiple tabs that search selected text and some tabs that will be a blank tab search.
Adjusted Mouse wheel sensitivity and had same results.
Turned off other add-ons but no change
Turned fission.autostart  to false but no change.

**Comment by Robbendebiene** [comment]
The bug has been fixed in Firefox 148

See:
- https://bugzilla.mozilla.org/show_bug.cgi?id=1741694#c25
- https://bugzilla.mozilla.org/show_bug.cgi?id=543435
